### PR TITLE
Allow resetting database from password view.

### LINKED ISSUE #146: Allow deleting all data on password dialog.
Currently if a user has set a password then the password dialog is shown at the start.

In the case that the user forgets this password, there is no way for them to completely reset their database other than manually navigating to the folder storing the info and deleting that.

I propose adding a button on the password dialog which allows the user to delete the db, starting completely from scratch.

Thoughts @sachaaaaa @BeaudanBrown @KeeJef

### LINKED ISSUE #1312: Allow deleting all data on password dialog.
Currently if a user has set a password then the password dialog is shown at the start.

In the case that the user forgets this password, there is no way for them to completely reset their database other than manually navigating to the folder storing the info and deleting that.

I propose adding a button on the password dialog which allows the user to delete the db, starting completely from scratch.

Thoughts @sachaaaaa @BeaudanBrown @KeeJef

### PR DESCRIPTION
**Pull request by Mikunj**
This PR allows the users to reset their database after they fail unlocking the app after 3 tries.

resolves oxen-io/session-desktop-temp#1202 

screen shot 2019-01-29 at 12 49 52 pm
screen shot 2019-01-29 at 12 49 59 pm

### CODE REVIEW

File: app/sql.js [hunk 1]
```diff
@@ -643,6 +643,15 @@ let db;
 let filePath;
 let indexedDBPath;
 
+function _initializePaths(configDir) {
```
[the remark this answers is not in the document]
Comment (Mikunj) [reply app/sql.js]: just indicates that it's a private function

File: js/views/clear_data_view.js [hunk 2]
```diff
@@ -21,8 +21,9 @@
       'click .cancel': 'onCancel',
       'click .delete-all-data': 'onDeleteAllData',
     },
-    initialize() {
+    initialize(onClear = null) {
```
[the remark this answers is not in the document]
Comment (Mikunj) [reply js/views/clear_data_view.js]:
```js
const clearDataView = new window.Whisper.ClearDataView(() => {
        window.resetDatabase();
      });
```
in `password_view`

File: app/sql.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (sachaaaaa) [reply app/sql.js]: ah of course, ta!

File: js/views/clear_data_view.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sachaaaaa) [reply js/views/clear_data_view.js]: Oh ok, the initialize function is called as a constructor :)

[the remark this answers is not in the document]
Comment (Mikunj) [reply js/views/clear_data_view.js]: Ye, this is how you do them for backbone views

**Comment by sachaaaaa** [comment]
Looks good!
Can you just remove the bits about linking with the phone again in the "delete all data" message?

**Review by sachaaaaa — APPROVED** [review]
Sweet as! Let's get this in.

**Pull request merged**